File: go/solutions/2038-remove-colored-pieces-if-both-neighbors-are-the-same-color.go
package solutions

func WinnerOfGame(colors string) bool {
	aCount, bCount := 0, 0
	for i := 1; i < len(colors)-1; i++ {
		if colors[i-1] == colors[i] && colors[i] == colors[i+1] {
			if colors[i] == 'A' {
				aCount++
			} else {
				bCount++
			}
		}
	}

	return aCount > bCount
}

File: js/2038-remove-colored-pieces-if-both-neighbors-are-the-same-color.js
/**
 * @param {string} colors
 * @return {boolean}
 */
var winnerOfGame = function (colors) {
  var aCount = 0,
    bCount = 0;
  for (var i = 1; i < colors.length - 1; i++) {
    if (
      colors.charCodeAt(i - 1) === colors.charCodeAt(i) &&
      colors.charCodeAt(i) === colors.charCodeAt(i + 1)
    ) {
      if (colors.charCodeAt(i) === "A".charCodeAt(0)) aCount++;
      else bCount++;
    }
  }

  return aCount > bCount;
};

module.exports = { winnerOfGame };
